gateway-ha/src/main/java/io/trino/gateway/proxyserver/ProxyRequestHandler.java

```diff
@@ -181,6 +183,8 @@ private void performRequest(
         ImmutableList.Builder<NewCookie> cookieBuilder = ImmutableList.builder();
         cookieBuilder.addAll(getOAuth2GatewayCookie(remoteUri, servletRequest));
 
+        requestBuilder.addHeader("X-Routing-Group", routingDestinationInfo.group());
```

Comment:
Can you pass the routing info directly to `recordBackendForQueryId` in the `future.transform` call? If not, then use `ROUTING_GROUP_HEADER` from `RoutingGroupSelector` as the header key